### PR TITLE
Initial deprecation of kubeadm v1beta1 apis

### PR DESCRIPTION
**Pull request by Klaven**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
This PR deprecates the v1beta1 apies in kubeadm in favor of the v1beta2 apis.
**Which issue(s) this PR fixes**:

Fixes #

There is no issue filed, was discussed in 18th September 2019 kubeadm office hours when version 1.17 was planned.

**Special notes for your reviewer**:
I hope this is going on the right path, I looked up `/api/apps/v1beta1` was deprecated and kind of just followed along. Please let me know anything else that needs done and I will get it done. Thank you!

**Does this PR introduce a user-facing change?**:

```release-note
Action Required: `kubeadm.k8s.io/v1beta1` has been deprecated, you should update your config to use newer non-deprecated API versions.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go [hunk 1]
```diff
@@ -113,6 +115,7 @@ type ClusterConfiguration struct {
 	ClusterName string `json:"clusterName,omitempty"`
 }
 
+// DEPRECATED
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go]: i think we should add the message:
```
DEPRECATED - This group version of X is deprecated by apis/kubeadm/v1beta2/X.
```

to:

InitConfiguration
JoinConfiguration
ClusterConfiguration

but not have it for the other objects.

**Comment by neolit123** [comment]
in the release note:

> you should update to use the `kubeadm/app/apis/kubeadm/v1beta2`

please omit `the`.

**Review by rosti — COMMENTED** [review]
Thanks @Klaven !
We also need to modify `validateSupportedVersion` (can be found [here](https://github.com/kubernetes/kubernetes/blob/974f201363b1cb3c3e1943722f10fe79bce4f576/cmd/kubeadm/app/util/config/common.go#L51)) in the following ways:
- Remove the `allowDeprecated` param and consider it always `false`.
- Add the v1beta1 version string to the (now empty) map of deprecated versions.
- Instead of returning an error, print a warning, that the version is deprecated and users need to use `kubeadm config migrate` to switch to latest.

**Comment by Klaven** [comment]
>     * Remove the `allowDeprecated` param and consider it always `false`.


@rosti I am not sure I agree with this. If I did it would let deprecated api be used with init and join. I split the function so that it only returned an error if not allowed, and if allowed it logs a warning. Now that I think about it... it really is not needed anyhow as when you are allowed to use it you are running migrate anyhow and the warning is just going to tell you the same. What do you think?

**Comment by rosti** [comment]
@Klaven the idea is to have beta level APIs working out of the box. No matter if they are deprecated or not. Hence, we need to log a warning.
However, I think, that you are right for the flag. Let's keep it that way, just change the error return into a `klog.Warning`.

File: cmd/kubeadm/app/util/config/common.go [hunk 2]
```diff
@@ -72,7 +74,7 @@ func validateSupportedVersion(gv schema.GroupVersion, allowDeprecated bool) erro
 	}
 
 	if _, present := deprecatedAPIVersions[gvString]; present && !allowDeprecated {
-		return errors.Errorf("your configuration file uses a deprecated API spec: %q. Please use 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.", gv.String())
+		klog.Errorf("your configuration file uses a deprecated API spec: %q. Please use 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.", gv.String())
```
Comment (rosti) [inline cmd/kubeadm/app/util/config/common.go]: s/Errorf/Warningf
Furthermore, no need to call `gv.String()`, just pass `gv`, `.String()` will be called under the hood.

**Comment by rosti** [comment]
Another thing, can you change the release note to something like this:
```
Action Required: `kubeadm.k8s.io/v1beta1` has been deprecated, you should update your config to use newer non-deprecated API versions.
```

**Review by rosti — APPROVED** [review]
Thanks @Klaven!
/approve
/lgtm

**Pull request merged**